qiskit/transpiler/passes/routing/algorithms/bip_model.py

```diff
@@ -0,0 +1,393 @@
+                mdl.add_constraint(z[t] >= z[t + 1], ctname=f"dummy_precedence_{t}")
+        # Symmetry breaking on the line -- only works on line topology!
+        if line_symm:
+            for h in range(1, self.num_vqubits):
+                mdl.add_constraint(
+                    sum(w[0, p, 0] for p in range(h))
+                    + sum([w[0, q, self.num_pqubits - 1] for q in range(h, self.num_vqubits)])
+                    >= 1,
+                    ctname=f"sym_break_line_{h}",
+                )
+
+        # *** Define objevtive function ***
+        if objective == "depth":
+            objexr = sum(z[t] for t in range(self.depth) if self._is_dummy_step(t))
+            for t in range(self.depth - 1):
+                for q in range(self.num_vqubits):
+                    for (i, j) in self._arcs:
+                        objexr += 0.01 * x[t, q, i, j]
+            mdl.minimize(objexr)
+        elif objective == "error_rate":
+            self._set_error_rate_obj(mdl)
+        elif objective == "balanced":
+            self._set_balanced_obj(mdl)
+        else:
+            raise TranspilerError(f"Unknown objective type: {objective}")
+
+        self.problem = mdl
+        logger.info("BIP problem stats: %s", self.problem.statistics)
+
+    @staticmethod
+    def _set_error_rate_obj(model):
+        """Set the minimum error rate objective function."""
+        raise NotImplementedError("objective: 'error_rate' is not implemented")
+
+    @staticmethod
+    def _set_balanced_obj(model):
+        """Set the minimum balanced (weighted sum of error_rate and depth) objective function."""
+        raise NotImplementedError("objective: 'balanced' is not implemented")
+
+    def solve_cpx_problem(self, time_limit: float = 60, threads: int = None):
+        """Solve the BIP problem using CPLEX.
+
+        Args:
+            time_limit:
+                Time limit (seconds) given to CPLEX.
+
+            threads:
+                Number of threads to be allowed for CPLEX to use.
+
+        Raises:
+            MissingOptionalLibraryError: If CPLEX is not installed
+            TranspilerError: if fails to solve the problem within given ``time_limit``.
+        """
+        if not HAS_CPLEX:
+            raise MissingOptionalLibraryError(
+                libname="CPLEX",
+                name="CplexOptimizer",
+                pip_install="pip install cplex",
+            )
+        self.problem.set_time_limit(time_limit)
+        if threads is not None:
+            self.problem.context.cplex_parameters.threads = threads
+        self.problem.context.cplex_parameters.randomseed = 777
+
+        self.solution = self.problem.solve()
+
+        status = self.problem.solve_details.status
+        logger.info("BIP solution status: %s", status)
+        if self.solution is None:
+            logger.warning("BIP solution status: %s", status)
+            raise TranspilerError("Failed to solve a BIP problem.")
```

Comment:
this exception is internal. I suggest to create your own exception to raise and catch.

Alternatively (an option that I personally favor), `solve_cpx_problem` could return the status and do the error handleling at `run` time:
```
        status = model.solve_cpx_problem(time_limit=self.time_limit, threads=self.threads)
        if self.solution is None:
            logger.warning("Failed to solve a BIP problem. Status: %s", status)
            return original_dag

Reply:
Fixed at b30359d